hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/supervisor/workload/ConsumerTestRuntimeEnvironment.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.commons.lang.math.RandomUtils;
+import org.apache.curator.framework.CuratorFramework;
+import pl.allegro.tech.hermes.api.Group;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException;
+import pl.allegro.tech.hermes.common.metric.HermesMetrics;
+import pl.allegro.tech.hermes.consumers.subscription.cache.SubscriptionsCache;
+import pl.allegro.tech.hermes.consumers.subscription.cache.zookeeper.ZookeeperSubscriptionsCacheFactory;
+import pl.allegro.tech.hermes.consumers.supervisor.ConsumersSupervisor;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.selective.ConsumerNodesRegistry;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.selective.SelectiveSupervisorController;
+import pl.allegro.tech.hermes.domain.group.GroupRepository;
+import pl.allegro.tech.hermes.domain.subscription.SubscriptionRepository;
+import pl.allegro.tech.hermes.domain.topic.TopicRepository;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperGroupRepository;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperPaths;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperSubscriptionRepository;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperTopicRepository;
+
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static com.jayway.awaitility.Duration.ONE_SECOND;
+import static org.apache.commons.lang.math.RandomUtils.nextInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static pl.allegro.tech.hermes.api.Topic.Builder.topic;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_REBALANCE_INTERVAL;
+import static pl.allegro.tech.hermes.test.helper.endpoint.TimeoutAdjuster.adjust;
+
+public class ConsumerTestRuntimeEnvironment {
+    private final ZookeeperPaths paths;
+    private final Supplier<CuratorFramework> curatorSupplier;
+    private GroupRepository groupRepository;
+    private TopicRepository topicRepository;
+    private SubscriptionRepository subscriptionRepository;
+    private MutableConfigFactory configFactory;
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private ConsumersSupervisor supervisor = mock(ConsumersSupervisor.class);
+    private static ExecutorService executorService = Executors.newSingleThreadExecutor();
+    private HermesMetrics metrics;
+    private ConsumerNodesRegistry consumersRegistry;
+    private CuratorFramework curator;
+
+    public ConsumerTestRuntimeEnvironment(Supplier<CuratorFramework> curatorSupplier) {
```

Comment:
This class is actually a side-effect of SelectiveSupervisorControllerIntegrationTest refactoring and I haven't finished yet. Thanks for your input though.